appengine/genetics/js/mouseAvatar.js

```diff
@@ -0,0 +1,307 @@
+  image.setAttribute('x', '0px');
+  if(this.sex === Genetics.Mouse.Sex.FEMALE) {
+    image.setAttribute('y', - (Genetics.MouseAvatar.HEIGHT + Genetics.MouseAvatar.TAIL_HEIGHT) + 'px');
+  }
+  image.style.transformOrigin = Genetics.MouseAvatar.HALF_SIZE + 'px ' + Genetics.MouseAvatar.HALF_SIZE + 'px'; // TODO are both transform origins necessary?
+  image.setAttribute('clip-path', 'url(#mouse' + mouse.id + 'ClipPath)');
+  this.element.appendChild(image);
+
+  // Calculate the pie chart arc start/end based on avatar size.
+  var xOffset = Genetics.MouseAvatar.WIDTH / 2 - Genetics.MouseAvatar.CHART_HALF_SIZE;
+  var yOffset = Genetics.MouseAvatar.HEIGHT * 4 / 5 - Genetics.MouseAvatar.CHART_HALF_SIZE;
+  var radius = Genetics.MouseAvatar.CHART_HALF_SIZE;
+  var x1 = radius + xOffset;
+  var y1 = yOffset;
+  var x2 = radius * (1 + 0.5 * Math.sqrt(3)) + xOffset;
+  var y2 = radius * 1.5 + yOffset;
+  var x3 = radius * (1 - 0.5 * Math.sqrt(3)) + xOffset;
+  var y3 = radius * 1.5 + yOffset;
+  var centerX = radius + xOffset;
+  var centerY = radius + yOffset;
+
+  // Draw top right slice.
+  var proposeMateSlice = document.createElementNS(Blockly.SVG_NS, 'path');
+  proposeMateSlice.setAttribute('d', 'M ' + x1 + ' ' + y1 +
+      ' A ' + radius + ' ' + radius + ', 0, 0, 1, ' +
+      x2 + ' ' + y2 + ' L ' + centerX + ' ' + centerY + ' Z');
+  proposeMateSlice.setAttribute('fill',
+      Genetics.Visualization.COLOURS[this.proposeMateOwner]);
+  this.element.appendChild(proposeMateSlice);
+
+  // Draw bottom slice.
+  var pickFightSlice = document.createElementNS(Blockly.SVG_NS, 'path');
+  pickFightSlice.setAttribute('d', 'M ' + x2 + ' ' + y2 +
+      ' A ' + radius + ' ' + radius + ', 0, 0, 1, ' +
+      x3 + ' ' + y3 + ' L ' + centerX + ' ' + centerY + ' Z');
+  pickFightSlice.setAttribute('fill',
+      Genetics.Visualization.COLOURS[this.pickFightOwner]);
+  this.element.appendChild(pickFightSlice);
+
+  // Draw top left slice.
+  var acceptMateSlice = document.createElementNS(Blockly.SVG_NS, 'path');
+  acceptMateSlice.setAttribute('d', 'M ' + x3 + ' ' + y3 +
+      ' A ' + radius + ' ' + radius + ', 0, 0, 1, ' +
+      x1 + ' ' + y1 + ' L ' + centerX + ' ' + centerY + ' Z');
+  acceptMateSlice.setAttribute('fill',
+      Genetics.Visualization.COLOURS[this.acceptMateOwner]);
+  this.element.appendChild(acceptMateSlice);
+
+  Object.defineProperty(this, 'direction', {
+    /**
+     * Sets the direction to a value between 0-2PI and rotates the mouse to
+     * match the direction facing. Direction angle is clockwise, with 0
+     * indicating the mouse facing right.
+     * @param {number} direction
+     */
+    set: function(direction) {
+      // Convert direction to a value between 0-2PI
+      while (direction < 0) {
+        direction += 2 * Math.PI;
+      }
+      while (direction > 2 * Math.PI) {
+        direction -= 2 * Math.PI;
+      }
```

Comment:
goog.math.standardAngleInRadians
Read through goog.math (which you already depend on) and see if there are other functions that you can use in this file.